File: packages/python/langchain_nuggets/toolkit.py
"""Nuggets toolkit providing all identity verification tools."""
from __future__ import annotations

import os
from typing import List, Optional

from langchain_core.tools import BaseTool

from langchain_nuggets.client.nuggets_api_client import NuggetsApiClient
from langchain_nuggets.tools.kyc import (
    InitiateKycVerification,
    CheckKycStatus,
    VerifyAge,
    VerifyCredential,
)
from langchain_nuggets.tools.kya import (
    RegisterAgentIdentity,
    VerifyAgentIdentity,
    GetAgentTrustScore,
)
from langchain_nuggets.tools.auth import (
    RequestCredentialPresentation,
    VerifyPresentation,
    InitiateOAuthFlow,
    CheckAuthStatus,
)


class NuggetsToolkit:
    """Toolkit that provides all Nuggets identity verification tools.

    Config can be passed directly or read from environment variables:
    - NUGGETS_API_URL
    - NUGGETS_PARTNER_ID
    - NUGGETS_PARTNER_SECRET
    """

    def __init__(
        self,
        api_url: Optional[str] = None,
        partner_id: Optional[str] = None,
        partner_secret: Optional[str] = None,
    ) -> None:
        resolved_api_url = api_url or os.environ.get("NUGGETS_API_URL", "")
        resolved_partner_id = partner_id or os.environ.get("NUGGETS_PARTNER_ID", "")
        resolved_partner_secret = partner_secret or os.environ.get("NUGGETS_PARTNER_SECRET", "")

        if not resolved_api_url or not resolved_partner_id or not resolved_partner_secret:
            raise ValueError(
                "NuggetsToolkit requires api_url, partner_id, and partner_secret. "
                "Provide them via constructor or NUGGETS_API_URL, NUGGETS_PARTNER_ID, "
                "NUGGETS_PARTNER_SECRET environment variables."
            )

        self._client = NuggetsApiClient({
            "api_url": resolved_api_url,
            "partner_id": resolved_partner_id,
            "partner_secret": resolved_partner_secret,
        })

    def get_tools(self) -> List[BaseTool]:
        """Return all 11 Nuggets identity verification tools."""
        params = {"client": self._client}
        return [
            InitiateKycVerification(**params),
            CheckKycStatus(**params),
            VerifyAge(**params),
            VerifyCredential(**params),
            RegisterAgentIdentity(**params),
            VerifyAgentIdentity(**params),
            GetAgentTrustScore(**params),
            RequestCredentialPresentation(**params),
            VerifyPresentation(**params),
            InitiateOAuthFlow(**params),
            CheckAuthStatus(**params),
        ]

File: packages/js/src/toolkit.ts
import type { StructuredToolInterface } from "@langchain/core/tools";
import { NuggetsApiClient } from "./client/nuggets-api-client.js";
import type { NuggetsConfig } from "./types.js";
import {
  InitiateKycVerification,
  CheckKycStatus,
  VerifyAge,
  VerifyCredential,
} from "./tools/kyc/index.js";
import {
  RegisterAgentIdentity,
  VerifyAgentIdentity,
  GetAgentTrustScore,
} from "./tools/kya/index.js";
import {
  RequestCredentialPresentation,
  VerifyPresentation,
  InitiateOAuthFlow,
  CheckAuthStatus,
} from "./tools/auth/index.js";

declare const process: { env: Record<string, string | undefined> };

export class NuggetsToolkit {
  private client: NuggetsApiClient;

  constructor(config?: Partial<NuggetsConfig>) {
    const resolvedConfig: NuggetsConfig = {
      apiUrl: config?.apiUrl || process.env.NUGGETS_API_URL || "",
      partnerId: config?.partnerId || process.env.NUGGETS_PARTNER_ID || "",
      partnerSecret: config?.partnerSecret || process.env.NUGGETS_PARTNER_SECRET || "",
      webhook: config?.webhook,
    };

    if (!resolvedConfig.apiUrl || !resolvedConfig.partnerId || !resolvedConfig.partnerSecret) {
      throw new Error(
        "NuggetsToolkit requires apiUrl, partnerId, and partnerSecret. " +
        "Provide them via constructor or NUGGETS_API_URL, NUGGETS_PARTNER_ID, " +
        "NUGGETS_PARTNER_SECRET environment variables."
      );
    }

    this.client = new NuggetsApiClient(resolvedConfig);
  }

  getTools(): StructuredToolInterface[] {
    const params = { client: this.client };
    return [
      new InitiateKycVerification(params),
      new CheckKycStatus(params),
      new VerifyAge(params),
      new VerifyCredential(params),
      new RegisterAgentIdentity(params),
      new VerifyAgentIdentity(params),
      new GetAgentTrustScore(params),
      new RequestCredentialPresentation(params),
      new VerifyPresentation(params),
      new InitiateOAuthFlow(params),
      new CheckAuthStatus(params),
    ];
  }
}
